libs/barista-components/overlay/src/overlay-trigger.spec.ts

```diff
@@ -145,6 +145,55 @@ describe('DtOverlayTrigger', () => {
     expect(overlay).not.toBeNull();
   }));
 
+  it('should fire pinnedChanged when pinned', fakeAsync(() => {
+    fixture.componentInstance.config = { pinnable: true };
+    fixture.detectChanges();
+    initOverlay(fixture, trigger);
+
+    dispatchMouseEvent(trigger, 'click');
+    fixture.detectChanges();
+    flush();
+
+    expect(fixture.componentInstance.pinned).toBeTruthy();
+  }));
+
+  it('should not fire pinnedChanged on subsequent mouseenter', fakeAsync(() => {
+    fixture.componentInstance.config = { pinnable: true };
+    fixture.detectChanges();
+    initOverlay(fixture, trigger);
+
+    dispatchMouseEvent(trigger, 'click');
+    fixture.detectChanges();
+    flush();
+
+    expect(fixture.componentInstance.pinned).toBeTruthy();
+
+    dispatchMouseEvent(trigger, 'mouseleave');
+    dispatchMouseEvent(trigger, 'mouseenter');
+    dispatchMouseEvent(trigger, 'mousemove');
+    fixture.detectChanges();
+
+    expect(fixture.componentInstance.pinned).toBeTruthy();
+  }));
+
+  it('should fire pinnedChanged when pinned then overlay is dismissed', fakeAsync(() => {
```

Comment:
add spies

Reply:
🥷

Reply:
actually no need for a spy. testing wether pinned changed in the TestComponent is sufficient